packages/hadron-document/src/document.ts

```diff
@@ -21,14 +21,22 @@ const ID = '_id';
 /**
  * Represents a document.
  */
-class Document extends EventEmitter {
+export class Document extends EventEmitter {
+  uuid: string;
+  doc: BSONObject;
+  cloned: boolean;
+  isUpdatable: boolean;
+  elements: LinkedList;
+  type: 'Document' | 'Array';
+  currentType: 'Document' | 'Array';
```

Comment:
I'm probably missing something, but can't find a place where this can be set to 'Array', or are we doing this to make it easier to work with in other parts of the code?

Reply:
Hm yeah … there was a check in this file that handled the `Array` case, but I don’t think that that was reachable (if it were, it seems like a lot of other assumptions would be broken) – I tried to keep the non-conversion changes minimal here, but you’re right, this is probably good to take care of now :slightly_smiling_face: Pushed a change!